From 9832663db6c175d2234743620e9f2cb41824e4ab Mon Sep 17 00:00:00 2001
From: Robert Nelson <robertcnelson@gmail.com>
Date: Thu, 30 Apr 2020 16:21:10 -0500
Subject: [PATCH 2/3] Revert "of: unittest: kmemleak in
 of_unittest_platform_populate()"

This reverts commit b712f00a98c10b6bca4f89cbe5cd04189b1dae7a.
---
 drivers/of/unittest.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/drivers/of/unittest.c b/drivers/of/unittest.c
index 7fde62484323..e285c0b04a9f 100644
--- a/drivers/of/unittest.c
+++ b/drivers/of/unittest.c
@@ -940,13 +940,10 @@ static void __init of_unittest_platform_populate(void)
 
 	of_platform_populate(np, match, NULL, &test_bus->dev);
 	for_each_child_of_node(np, child) {
-		for_each_child_of_node(child, grandchild) {
-			pdev = of_find_device_by_node(grandchild);
-			unittest(pdev,
+		for_each_child_of_node(child, grandchild)
+			unittest(of_find_device_by_node(grandchild),
 				 "Could not create device for node '%s'\n",
 				 grandchild->name);
-			of_dev_put(pdev);
-		}
 	}
 
 	of_platform_depopulate(&test_bus->dev);
-- 
2.26.2

